src/Model/ElementDecisionTree.php

```diff
@@ -75,4 +76,89 @@ public function CMSEditFirstStepLink()
             $this->FirstStep()->ID
         );
     }
+
+    /**
+     * To provide all the content block data as JSON to make javascript
+     * based front ends easier to build.
+     *
+     * @return string
+     */
+    public function getBlockJSON()
+    {
+        return json_encode(array_merge(
+            [
+                'blockTitle' => $this->Title,
+                'blockIntro' => ShortcodeParser::get('default')->parse($this->Introduction),
+            ],
+            $this->getTreeData()
+        ));
+    }
+
+    /**
+     * Helper to extract the step and answer data from the block.
+     *
+     * @return array
+     */
+    private function getTreeData()
+    {
+        $data = [
+            'steps' => [],
+            'answers' => [],
+        ];
+
+        // start from the top
+        $first = $this->FirstStep();
+
+        if (!$first->exists()) {
+            return $data;
+        }
+
+        // now start our descent through the flow
+        $this->collectStepData($first, $data);
+
+        // provide way to inject extra data
+        $this->extend('updateTreeData', $data);
+
+        return $data;
+    }
+
+    /**
+     * Helper to collect the data for a given step.
+     *
+     * @param DecisionTreeStep $step
+     * @param array &$data
+     * @return void
+     */
+    private function collectStepData($step, &$data)
+    {
+        // add step data into array
+        $data['steps'][$step->ID] = $step->toJSONData();
```

Comment:
I'm aware this suggestion might mess up any existing implementations of this solution; but given that this is intended for JS, is it worthwhile not using an index here, and instead allow JS to parse this into an array? An object of index properties feels like a less predictable js object structure.

Reply:
Found this was the better approach as it is faster and easier when it comes to finding the data for the next step/answer to render, e.g. you loop to find the first step to render, that gives you `answers` as an array of ids for the answers data, so you can then just extract out the answer data referencing those ids as keys from the `answers` data vs it being an array and having to loop the whole answers array multiple times to find the data you are looking for.